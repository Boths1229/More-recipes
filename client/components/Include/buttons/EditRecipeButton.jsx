import React from 'react';
import PropTypes from 'prop-types';

/**
 * @description function component to render edit recipe button
 *
 * @param {object} props
 *
 * @returns {JSX} return JSX
 */
const EditRecipeButton = props => ((
  <div>
    <button
      type="button"
      className="btn"
      data-toggle="modal"
      data-target="#editModal"
      data-whatever="@getbootstrap"
      onClick={props.setEditRecipeId}
    >
      <i className="fa fa-pencil-square-o" aria-hidden="true" />
    </button>
  </div>
));

EditRecipeButton.propTypes = {
  setEditRecipeId: PropTypes.func.isRequired
};

export default EditRecipeButton;
