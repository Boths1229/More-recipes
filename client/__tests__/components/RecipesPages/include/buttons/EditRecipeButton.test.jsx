import React from 'react';
import { shallow } from 'enzyme';
import EditRecipeButton from '../../../../../components/Include/buttons/EditRecipeButton';

describe('Edit recipe button component', () => {
  it('matches snapshot', () => {
    const output = shallow(<EditRecipeButton />);
    expect(output).toMatchSnapshot();
  });
  it('calls onClick', () => {
    const props = {
      value: {},
      onEditClicked: jest.fn()
    };
    const output = shallow(<EditRecipeButton {...props} />);
    const button = output.find('button[type="button"]');
    button.simulate('click');
    expect(props.onEditClicked).toHaveBeenCalled();
  });
});
