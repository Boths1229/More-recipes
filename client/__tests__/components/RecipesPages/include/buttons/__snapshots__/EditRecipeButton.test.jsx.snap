// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Edit recipe button component matches snapshot 1`] = `
<div>
  <button
    className="btn edit-button"
    data-target="#editModal"
    data-toggle="modal"
    data-whatever="@getbootstrap"
    type="button"
  >
    <i
      aria-hidden="true"
      className="fa fa-pencil-square-o"
    />
  </button>
</div>
`;
